File: POO JAVA/Aula08/src/aula08/Luta.java
package aula08;

import java.util.Random;

public class Luta {

    //Atributos da Classe
    private Lutador desafiante;
    private Lutador desafiado;
    private int rounds;
    private boolean aprovada;

    //Metodo Construtor
    public Luta() {    
        this.rounds = 0;
        this.aprovada = false;
    }

    //Metodos Acessores e Modificadores
    public Lutador getDesafiante() {
        return desafiante;
    }
    public void setDesafiante(Lutador desafiante) {
        this.desafiante = desafiante;
    }
    public Lutador getDesaficado() {
        return desafiado;
    }
    public void setDesaficado(Lutador desaficado) {
        this.desafiado = desaficado;
    }
    public int getRounds() {
        return rounds;
    }
    public void setRounds(int rounds) {
        this.rounds = rounds;
    }
    public boolean getAprovada() {
        return aprovada;
    }
    public void setAprovada(boolean aprovada) {
        this.aprovada = aprovada;
    }
    
    //Metodos da Classe
    public void marcarLuta(Lutador L1, Lutador L2){
        if(L1.getCategoria().equals(L2.getCategoria()) && L1 != L2){
            this.aprovada = true;
            this.desafiante = L1;
            this.desafiado = L2;
            System.out.println("\nA lutar foi MARCADA !!!");
        }else{
            System.out.println("\nLuta nao MARCADO, lutadores de categoria diferentes.");
        }
    }
    public void lutar(){
        if(this.aprovada){
            System.out.print("\n## DESAFIANTE ##");
            System.out.println(this.desafiante.apresentar());
            System.out.print("\n## DESAFIADO ##");
            System.out.println(this.desafiado.apresentar());
            
            Random ale = new Random();
            int a = ale.nextInt(3); //1, 2, 3
            this.rounds = ale.nextInt(4) + 1;
            switch(a){
                case 0: //Empate
                    System.out.println("\nRound: "+this.rounds);
                    System.out.println("EMPATOU");
                    this.desafiante.empatarLuta();
                    this.desafiado.empatarLuta();
                    break;
                case 1: //Vitoria do desafiante
                    System.out.println("\nRound: "+this.rounds);
                    System.out.println("Vitoria do DESAFIANTE");
                    this.desafiante.ganharLuta();
                    this.desafiado.perderLuta();
                    break;
                case 2: //Vitoria do desafiado
                    System.out.println("\nRound: "+this.rounds);
                    System.out.println("Vitoria do DESAFIADO");
                    this.desafiante.perderLuta();
                    this.desafiado.ganharLuta();
                    break;
            }
        }else{
            System.out.println("Erro !!!\nLuta nao marcada ou nao aprovada.");
        }
    }
}

File: POO PHP/Aula08/Luta.php
<?php

require_once 'Lutador.php';

class Luta {

    //Atributos da Classe
    private $desafiante;
    private $desafiado;
    private $round;
    private $aprovada;
    
    //Metodos Acessores e Modificadores
    public function getDesafiante() {
        return $this->desafiante;
    }
    public function getDesafiado() {
        return $this->desafiado;
    }
    public function getRound() {
        return $this->round;
    }
    public function getAprovada() {
        return $this->aprovada;
    }
    public function setDesafiante($desafiante): void {
        $this->desafiante = $desafiante;
    }
    public function setDesafiado($desafiado): void {
        $this->desafiado = $desafiado;
    }
    public function setRound($round): void {
        $this->round = $round;
    }
    public function setAprovada($aprovada): void {
        $this->aprovada = $aprovada;
    }

    //Metodos da Classe
    public function marcarLuta($ludador1, $lutador2) {
        if($ludador1->getCategoria() === $lutador2->getCategoria() && $ludador1 != $lutador2){
            $this->aprovada = true;
            $this->desafiante = $ludador1;
            $this->desafiado = $lutador2;
            echo "Luta Marcada pra Aconter Hoje, preparem os ringues !!!";
        } else {
            $this->aprovada = false;
            $this->desafiante = null;
            $this->desafiado = null;
            echo "Erro !!! Lutadores de categoria diferente";
        }
    }
    public function lutar() {
        if($this->aprovada){
            $this->desafiante->apresentar();
            $this->desafiado->apresentar();
            $venc = rand(0, 2);
            switch ($venc) {
                case 0: //Empate
                    echo "<br><p>Empate !</p>";
                    $this->desafiante->empatarLuta();
                    $this->desafiado->empatarLuta();
                    break;
                case 1: //Desafiante Ganha
                    echo "<br><p>Desafiante ".$this->desafiante->getNome()." Venceu !</p>";
                    $this->desafiante->ganharLuta();
                    $this->desafiado->perderLuta();
                    break;
                case 2:
                    echo "<br><p>Desafiante ".$this->desafiado->getNome()." Venceu !</p>";
                    $this->desafiante->perderLuta();
                    $this->desafiado->ganharLuta();
                    break;
            }
            
        } else {
            echo "Luta não pode acontecer";
        }
    }

}
